CombatCrittersSharp/objects/card/CardQueryBuilder.cs

```diff
@@ -1,36 +1,82 @@
 using CombatCrittersSharp.objects.card.Interfaces;
+using static CombatCrittersSharp.objects.card.Interfaces.ICardQuery;
 
 namespace CombatCrittersSharp.objects.card
 {
     public class CardQueryBuilder : ICardQueryBuilder
     {
-        private Dictionary<string, string> _queryParameters = new();
+        private int _cost;
+        private bool _costLess;
+        private List<int> _ids;
+        private CardOrder _order;
+        private bool _owned;
+        private List<int> _rarities;
+        private bool _raritiesInclude;
+        
 
-        public ICardQueryBuilder ByType(string type)
+        //constructor
+        public CardQueryBuilder()
         {
-            _queryParameters["type"] = type;
-            return this;
+            //Reset the values
+            Reset();
         }
 
-        public ICardQueryBuilder ByRarity(int rarity)
+        //Builds the final query object
+        public ICardQuery Build()
         {
-            _queryParameters["rarity"] = rarity.ToString();
-            return this;
+            return new CardQuery(
+                _cost,
+                _costLess,
+                _ids,
+                _order,
+                _owned,
+                _rarities,
+                _raritiesInclude
+            );
         }
 
-        public string GetQueryString()
+        public void Reset()
+        {
+            _cost = 0;
+            _costLess = true;
```

Comment:
cost less should default false otherwise by default you will get nothing from the query

Reply:
Okay